fem/gridfunc.hpp

```diff
@@ -755,176 +755,6 @@ class JumpScaling
    }
 };
 
-
-/** @brief Class representing a function through its values (scalar or vector)
-    at quadrature points. */
-class QuadratureFunction : public Vector
-{
-protected:
-   QuadratureSpace *qspace; ///< Associated QuadratureSpace
-   int vdim;                ///< Vector dimension
-   bool own_qspace;         ///< QuadratureSpace ownership flag
-
-public:
-   /// Create an empty QuadratureFunction.
-   /** The object can be initialized later using the SetSpace() methods. */
-   QuadratureFunction()
-      : qspace(NULL), vdim(0), own_qspace(false) { }
-
-   /** @brief Copy constructor. The QuadratureSpace ownership flag, #own_qspace,
-       in the new object is set to false. */
-   QuadratureFunction(const QuadratureFunction &orig)
-      : Vector(orig),
-        qspace(orig.qspace), vdim(orig.vdim), own_qspace(false) { }
-
-   /// Create a QuadratureFunction based on the given QuadratureSpace.
-   /** The QuadratureFunction does not assume ownership of the QuadratureSpace.
-       @note The Vector data is not initialized. */
-   QuadratureFunction(QuadratureSpace *qspace_, int vdim_ = 1)
-      : Vector(vdim_*qspace_->GetSize()),
-        qspace(qspace_), vdim(vdim_), own_qspace(false) { }
-
-   /** @brief Create a QuadratureFunction based on the given QuadratureSpace,
-       using the external data, @a qf_data. */
-   /** The QuadratureFunction does not assume ownership of neither the
-       QuadratureSpace nor the external data. */
-   QuadratureFunction(QuadratureSpace *qspace_, double *qf_data, int vdim_ = 1)
-      : Vector(qf_data, vdim_*qspace_->GetSize()),
-        qspace(qspace_), vdim(vdim_), own_qspace(false) { }
```

Comment:
@pazner , it looks like this constructor was not copied to the new file, `qfunction.?pp`. Was there a reason to remove it, or was this by accident. This was noticed here: https://github.com/LLNL/axom/pull/975#discussion_r1044844974.

Reply:
Looks like it was by accident.

Would an interface based on `Memory` make more sense given `QuadratureFunction` usually lives on device?

Reply:
Okay, just wanted to check, in case you had some reason to remove it.

In this constructor, the assumption is that the pointer is a host pointer. If we want to add `Memory`-based version that will support device pointers, we can add that. I think @kennyweiss just wanted to restore this one -- I think he was planning to make a PR for that in the near future.

Reply:
Thanks -- https://github.com/mfem/mfem/pull/3362 adds back the missing constructor and I tested it against my axom branch from https://github.com/LLNL/axom/pull/975